File: demo-iys-rest-scraping.java/src/main/java/demo/js/SourceListSerializer.java
package demo.js;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.reflect.TypeToken;
import demo.model.Source;

import java.lang.reflect.Type;
import java.util.List;

class SourceListSerializer {

    private static final Gson GSON = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
    private static final Type TYPE = new TypeToken<List<Source>>() {}.getType();

    static List<Source> fromJson(String json) {
        return GSON.fromJson(json, TYPE);
    }
}

File: demo-iys-rest-scraping.kotlin/src/main/kotlin/demo/js/SourceListSerializer.kt
package demo.js

import com.google.gson.GsonBuilder
import com.google.gson.reflect.TypeToken
import demo.model.Source

internal object SourceListSerializer {

    private val GSON = GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create()
    private val TYPE = object : TypeToken<List<Source>>() {
    }.type

    fun fromJson(json: String): List<Source> {
        return GSON.fromJson(json, TYPE)
    }
}
